### ISSUE TITLE
Backup the WTF folder

### ISSUE THREAD

**Issue by Jeppeskovsgaard** [jliddev/WowUp#22]
A couple of weeks a go, my computer crashed and when I logged back into wow, all settings were reset. I tried using the .bak files instead, but with no luck. I had to start over...

A Google search revealed that many other players have experienced exactly the same issue.

I don't know if this is out of scope, but an optional backup feature of the addon settings (WTF folder) would be nice. Either automatically when this application is opened or manually by pressing a button.

**Comment by jliddev** [comment]
Just to brainstorm, you'd want to back up your settings locally just in case?

**Comment by Jeppeskovsgaard** [comment]
Yes, exactly. Then I would be able to restore my addon settings in case they were reset again.

Just brainstorming now; thinking ahead, the backup feature could be expanded to enable users to export all addons and settings. Many players are asking other players or streamers about their addons and settings. With an export feature it would be super easy to share everything.

**Comment by sampaiodias** [comment]
Going further with the ideas, I think there are so many possibilities for this feature that a we should talk about what a MVP would look like. For example, simply having a button to copy the contents of the WTF folder to a subfolder (e.g.: C:/FolderTheUserSelected/retail-addon-backup-2020-11-07-10-08-35). Instead of copying the AddOns folder, backing up a list of them should enough (download them again if needed).

Giving users some settings would also be desirable:

1. Choose a destination folder for your backups (for each client) 
2. Choose how many backups you can create. If the maximum amount of backups are reached, the oldest backup is deleted (upon completion) to preserve disk space.

Restoring a backup COULD be part of the MVP or not, as the backup itself is the most import part (restoration will rarely be done and it's just a copy paste).

**Comment by nduitz** [comment]
Hey there,
I thought of this feature today, going a step further having an account that allows automated "cloud" synchronization of add-ons +  addon-profiles. I know that wouldn't be trivial and there is a lot to discuss:
How should the user service be provided?
 - third party provide e.g. auth0 ?
 - API backend? Where to host? Heroku? DigitalOcean?
Where should data be stored ?
- AWS S3
- DigitalOcean Spaces
- webserver
How is data protected?
- encryption obviously ;)

I for myself am not a fan of "as a service" providers. Sure managing a webserver + database + storage requires a bit of maintenance but I did a lot auf experiements with services like auth0 and I was never happy with the result.

I am experienced with rails/elixir-phoenix backends, I am part of team that manages a multinode multidatabase application on AWS and I am willing to develope an MVP as I see a great benefit from a feature like this.

I'd be happy to hear about your thoughts, my choices would be:
Rails API only backend (user management)
GitHub actions for CI
Passenger for deployment
PostgreSQL as a database
DigitalOcean Node (5$/mo) with caddy as a reverse proxy
Storage on DigitalOcean(5$/mo)

**Comment by Cassiopeia-the-Elf** [comment]
Hmm.. I am using v2.5.2 version and it has this feature: https://wowup.io/guide/my-addons/addon-settings Isn't it what OP is asking?

**Comment by linaori** [comment]
Yes, pretty much. In 2,5.0 a manual local backup feature was added as a start to see where we go from there

**Comment by Jiniix** [comment]
Would love a feature that implemented automatic backup of the WTF folder.
Either in conjunction with startup or when called with -m -q from the command line, or both. 
How it worked with CurseBreaker:
Run the update, program compresses WTF folder to a .zip named after timestamp and put it in WTF-backup located in your WoW root folder.
